modules/siddhi-core/src/main/java/io/siddhi/core/annotation/ProcessorAnnotationValidator.java

```diff
@@ -0,0 +1,141 @@
+            //validation here. Currently this logic is written assuming dynamic=true in @Parameter is same as
+            //instanceof VariableExpressionExecutor=true in init method
+            int m = -1;
+            boolean atleastOne = false;
+            boolean isCorrect;
+            for (ParameterOverload parameterOverload : parameterOverloads) {
+                isCorrect = true;
+                String[] parameterNames = parameterOverload.parameterNames();
+                if (parameterNames.length == attributeExpressionExecutors.length) {
+                    for (String parameterName : parameterNames) {
+                        if (isCorrect) {
+                            m++;
+                            for (Parameter parameter : parameters) {
+                                String parameterName1 = parameter.name();
+                                if (parameterName1.equals(parameterName)) {
+                                    boolean dynamic = parameter.dynamic();
+                                    boolean notDynamic = attributeExpressionExecutors[m]
+                                            instanceof ConstantExpressionExecutor;
+                                    isCorrect = (dynamic == !notDynamic);
+                                }
+                            }
+                        }
+                    }
+                    m = -1;
+                    if (isCorrect) {
+                        atleastOne = true;
+                    }
+                }
+            }
+            if (!atleastOne) {
+                throw new SiddhiAppValidationException("Input parameters dynamic property doesn't match");
+            }
+
+            //Check if data types of parameters match atleast one pattern specified in @ParameterOverload
+            int n = -1;
+            boolean atleastOnce = false;
+            boolean isCorrectOne;
+            for (ParameterOverload parameterOverload : parameterOverloads) {
+                isCorrectOne = true;
+                String[] parameterNames = parameterOverload.parameterNames();
+                if (parameterNames.length == attributeExpressionExecutors.length) {
+                    for (String parameterName : parameterNames) {
+                        if (isCorrectOne) {
+                            n++;
+                            boolean found = false;
+                            for (Parameter parameter : parameters) {
+                                String parameterName1 = parameter.name();
+                                if (parameterName1.equals(parameterName)) {
+                                    DataType[] types = parameter.type();
+                                    for (DataType type : types) {
+                                        if (attributeExpressionExecutors[n].getReturnType().toString().
+                                                equals(type.toString())) {
+                                            found = true;
```

Comment:
We don't need to continue the loop after finding the correct entry.

Reply:
Fixed in 8fd5a4f